packages/server/modules/core/helpers/userHelper.ts

```diff
@@ -25,3 +26,23 @@ export function removePrivateFields(
 }
 
 export type { LimitedUserRecord, UserRecord }
+
+export const getUsersBaseQuery = (
+  query: Knex.QueryBuilder,
+  { searchQuery, role }: { searchQuery: string | null; role: string | null }
+) => {
+  if (searchQuery) {
+    query.where((queryBuilder) => {
```

Comment:
I understand and agree with you.
The problem is that this particular function is used once in the repo and some services.
The name is also misleading.
For now, I am trying to change the least possible amount of logic this PR has already introduced many changes.

Ideally, this function should not even exist outside of the repository because composing queries outside of the repository doesn't make any sense.
Putting this function inside the repository and exporting it means I should also add a domain type for it which is incorrect since that function is a utility but has no domain meaning.

Reply:
> Putting this function inside the repository and exporting it means I should also add a domain type

Why? I don't see why that would be the case at all. Repository layer is for repository logic, that's all it means, there is no expectation that "everything in the repository layer must be of domain level importance"